addons/l10n_in_edi/models/account_edi_format.py

```diff
@@ -346,6 +353,40 @@ def _get_l10n_in_edi_line_details(self, index, line, line_tax_details, sign):
         }
 
     def _l10n_in_edi_generate_invoice_json(self, invoice):
+        def put_discount_on(discount_line_vals, other_line_vals):
+            discount = max_discount = discount_line_vals['AssAmt'] * -1
+            if other_line_vals['AssAmt'] < discount:
+                max_discount = other_line_vals['AssAmt']
+            ratio_of_discount = max_discount / discount
```

Comment:
need to be careful with this calculation because you could have an existing discount...

Reply:
"AssAmt" is after a discount so do not worry about existing discount